src/AsyncUtils.js

```diff
@@ -11,10 +11,10 @@ export async function* map<T, U>(
 }
 
 export async function* filter<T>(
-  iterable: AsyncIterable<T>,
+  iterable: Promise<AsyncIterable<T>>,
```

Comment:
this should take an async iterable though

Reply:
why can't it take a promise and await on it?

Reply:
could be `AsyncIterable<T> | Promise<AsyncIterable<T>>` so you can simply await it?

Reply:
or, well, I can probably `await` on the consumer side 🤔